packages/puppeteer-core/src/cdp/Page.ts

```diff
@@ -417,6 +417,12 @@ export class CdpPage extends Page {
     return this.#emulationManager.javascriptEnabled;
   }
 
+  override async openDevTools(): Promise<Page> {
+    const pageTargetId = this.target()._targetId;
+    const devtoolsPage = await this.browserContext().openDevTools(pageTargetId);
```

Comment:
Can we use `this.browser()` and skip browser context here? Let's also downcast to a CDP Browser and only implement  openDevTools on the CDP Browser.